app/src/test/java/org/mozilla/fenix/ext/ContextTest.kt

```diff
@@ -0,0 +1,107 @@
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import android.app.Activity
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.content.Intent.ACTION_SEND
+import android.content.Intent.EXTRA_SUBJECT
+import android.content.Intent.EXTRA_TEXT
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import android.view.ContextThemeWrapper
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
+import androidx.annotation.DimenRes
+import androidx.annotation.StringRes
+import androidx.fragment.app.FragmentActivity
+import mozilla.components.browser.search.SearchEngineManager
+import mozilla.components.support.base.log.Log
+import mozilla.components.support.base.log.Log.Priority.WARN
+import mozilla.components.support.ktx.android.content.getColorFromAttr
+import org.jetbrains.anko.dimen
+import org.jetbrains.anko.px2dip
+import org.mozilla.fenix.BuildConfig
+import org.mozilla.fenix.FenixApplication
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.Components
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.utils.Settings
+import mozilla.components.support.test.robolectric.testContext
+
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class ContextTest () {
+
+    @Test
+    fun `GIVEN context WHEN seeking application of context THEN send back application context`() {
+        val expectedAppValue = testContext.getApplicationContext() as FenixApplication
+        val resultAppValue = testContext.application
+        assertEquals(expectedAppValue, resultAppValue)
+    }
+
+    @Test
+    fun `GIVEN context WHEN requiring components THEN send back application components`() {
+        val expectedComponentsValue = (testContext.getApplicationContext() as FenixApplication).components
+        val resultComponentsValue = testContext.components
+        assertEquals(expectedComponentsValue, resultComponentsValue)
+    }
+
+    @Test
+    fun `GIVEN context WHEN getting metrics controller THEN send back metrics`() {
+        val expectedMetricsValue = (testContext.getApplicationContext() as FenixApplication).components.analytics.metrics
+        val resultMetricsValue = testContext.metrics
+        assertEquals(expectedMetricsValue, resultMetricsValue)
+    }
+
+    @Test
+    fun `GIVEN context WHEN make it an activity`() {
+        val expectedActivity = testContext as? Activity
+        val actualResult = testContext.asActivity()
+        assertEquals(expectedActivity, actualResult)
+    }
+
+    @Test
+    fun `GIVEN context WHEN make it a fragment activity`() {
+        val expectedActivity = testContext as? FragmentActivity
```

Comment:
This might compare null == null? I'm not sure if `testContext` is an instance of Activity at all.

Using mocks might be a bit more consistent, but I'm not sure:
```kt
val context: Context = mock<FragmentActivity>()
val actualResult = context.asFragmentActivity()
assertEquals(context, actualResult)
```